WordPress/gutenberg
# First stab at in-toolbar link editing

## Description

This PR adds `ToolbarLinkControl` as a default way of editing links on the experimental navigation screen (see https://github.com/WordPress/gutenberg/issues/23375).

The control wires together expanded BlockControls and reusable parts of LinkControl to provide in-toolbar link editing UI as on the screenshot below:

![2020-07-23 14-37-50 2020-07-23 14_39_05](https://user-images.githubusercontent.com/205419/88287234-4f21f980-ccf2-11ea-8a37-3aa18e3a4c59.gif)

## Caveats:

1. During editing it's really hard to tell which link block is currently selected - thoughts?
1. Some margins and paddings are imperfect, for example the dropdown menu button isn't perfectly centered. This seems to be a bug in the toolbar UI and is outside of scope of this PR.
1. The suggestions dropdown disappears when the settings dropdown shows up. Should it re-appear when the input gains focus again?
1. Keyboard navigation reflects the [prototype](https://codesandbox.io/s/toolbartextcontrol-cv82w?file=/src/App.js:43-125) created by @diegohaz and feels pretty natural! ~~Keyboard navigation is far from perfect and needs to be iterated on, but maybe it's not a blocker for merging the first iteration?~~
1. ~~At the moment link-related changes are only accepted if you explicitly confirm them by pressing the "Done" button inside the toolbar. If you select a different block, the changes are disregarded. To me, it feels natural with regard to the text field, but unnatural with regard to the dropdown menu - discussion appreciated.~~

## How has this been tested?

1. Insert a navigation block in the post editor, edit a link, confirm the UI (LinkControl) didn't change after applying the PR.
1. Edit a link on the experimental navigation screen, confirm you get the toolbar editor, confirm it's functional (aside of the keyboard a11y aspect). Confirm all the settings are reflected after opening the toolbar again. Saving `rel` and `opensInNewPage` attributes does not actually work, but it's unrelated to this PR.
1. Dropdown menu is pretty limited for now. The original design proposed a toggle inside a button - that's a completely new interaction so I went for toggle-able checkmarks instead.

## Types of changes
New feature (non-breaking change which adds functionality).

## Checklist:
